test/src/kbasesearchengine/test/main/SearchMethodsTest.java

```diff
@@ -420,9 +422,113 @@ private SearchObjectsOutput searchObjects(
+        final  IndexingStorage idx = mock(IndexingStorage.class);
+
+        GUID guid = new GUID("ws:1/2/3");
+        Instant time = Instant.now();
+
+        //highlight already tested seperately. TODO add parent GUID && object
+        final kbasesearchengine.search.ObjectData obj = kbasesearchengine.search.ObjectData
+                .getBuilder(guid)
+                .withNullableTimestamp(time)
+                .withNullableData("obj")
+                .withNullableObjectName("objname")
+                .withKeyProperty("key", "prop")
+                .withNullableType(new SearchObjectType("Blah" , 1))
+                .withNullableCreator("user")
+                .withNullableCopier("user2")
+                .withNullableModule("module")
+                .withNullableMethod("method")
+                .withNullableModuleVersion("2")
+                .withNullableCommitHash("commitHash")
+                .build();
+
+        final ArrayList<ObjectData> objs = new ArrayList<>();
+        objs.add(obj);
+
+        final SortingRule sr = new SortingRule();
+        sr.isTimestamp = true;
+        sr.ascending = true;
+
+        final kbasesearchengine.search.MatchFilter filter = kbasesearchengine.search.MatchFilter.getBuilder()
+                .withNullableFullTextInAll("objname")
+                .build();
+
+        final PostProcessing pp1 = new PostProcessing();
+        pp1.objectHighlight = false;
+        pp1.objectData = false;
+        pp1.objectKeys = false;
+        pp1.objectInfo = true;
+        pp1.objectDataIncludes=null;
+
+
+        final FoundHits fh1 = new FoundHits();
+        fh1.pagination = null;
+        fh1.sortingRules =  Arrays.asList(sr);
+        fh1.total = 1;
+        fh1.guids = set(new GUID("ws:1/2/3"));
+        fh1.objects = objs;
+
+        //result with highlight
+        when(idx.searchObjects(
+                new ArrayList<>(),
+                filter,
+                null, // sort
+                new kbasesearchengine.search.AccessFilter().withAccessGroups(set()),
+                null, // pagination
+                pp1))
+                .thenReturn(fh1);
+
+        //tests
+        //with highlight on only
+        kbasesearchengine.PostProcessing option1 = new kbasesearchengine.PostProcessing()
+                .withIncludeHighlight(0L)
+                .withSkipKeys(1L)
+                .withSkipInfo(0L)
+                .withSkipData(1L);
+
+        final SearchObjectsOutput res1 = searchObjects(idx, option1, "objname");
+
+        assertThat("did not find objects", res1.getObjects().size() == 1, is(true));
+        kbasesearchengine.ObjectData actual = res1.getObjects().get(0);
+
+        assertThat("incorrect data", actual.getData().toString(),
+                is(new UObject("obj").toString()));
+        assertThat("incorrect guid", actual.getGuid(), is(guid.toString()));
+        Map <String, String> expectedKeyProps = new HashMap<>();
+
+        expectedKeyProps.put("key", "prop");
```

Comment:
Skip keys was true, so shouldn't this be empty?

Reply:
ops intent was to turn on everything